src/Admin/AbstractAdmin.php

```diff
@@ -2292,32 +2181,32 @@ public function addExtension(AdminExtensionInterface $extension): void
         $this->extensions[] = $extension;
     }
 
-    public function getExtensions()
+    public function getExtensions(): array
     {
         return $this->extensions;
     }
 
-    public function setMenuFactory(FactoryInterface $menuFactory): void
+    public function setMenuFactory(?FactoryInterface $menuFactory): void
     {
         $this->menuFactory = $menuFactory;
     }
 
-    public function getMenuFactory()
+    public function getMenuFactory(): ?FactoryInterface
     {
         return $this->menuFactory;
     }
 
-    public function setRouteBuilder(RouteBuilderInterface $routeBuilder): void
+    public function setRouteBuilder(?RouteBuilderInterface $routeBuilder): void
     {
         $this->routeBuilder = $routeBuilder;
     }
 
-    public function getRouteBuilder()
+    public function getRouteBuilder(): ?RouteBuilderInterface
     {
         return $this->routeBuilder;
     }
 
-    public function toString($object)
+    public function toString($object): string
```

Comment:
```suggestion
    public function toString(object $object): string
```

And remove the is_object check.
This method should be called only on object ; and ATM our code only call toString on object.

Reply:
Please, see https://github.com/sonata-project/SonataAdminBundle/pull/6005#discussion_r419017194.